Extension/src/LanguageServer/ui_new.ts

```diff
@@ -0,0 +1,530 @@
+    }
+
+    private setIsParsingFiles(val: boolean): void {
+
+        this.isParsingFiles = val;
+        const showIcon: boolean = val || (!this.isParsingWorkspacePaused && this.isParsingWorkspace);
+        const twoStatus: boolean = val && this.isParsingWorkspace;
+
+        // Leave this outside for more realtime respone
+        this.browseEngineStatusBarItem.busy = showIcon;
+
+        if (showIcon) {
+            this.browseEngineStatusBarItem.text = "$(database)";
+            this.browseEngineStatusBarItem.detail = (val ? this.parsingFilesTooltip : "")
+                + (twoStatus ? " | " : "")
+                + (this.isParsingWorkspace ? this.workspaceParsingStatus : "");
+
+            if (this.dbTimeout) {
+                clearTimeout(this.dbTimeout);
+            }
+        } else if (!this.isParsingWorkspace && !val) {
+            this.dbTimeout = setTimeout(() => {
+                this.browseEngineStatusBarItem.text = this.workspaceParseingDoneText;
+                this.browseEngineStatusBarItem.detail = "";
+                this.browseEngineStatusBarItem.command = {
+                    command: "C_Cpp.RescanWorkspaceUI_Telemetry",
+                    title: this.workspaceRescanText
+                };
+            }, this.iconDelayTime);
+        }
+    }
+
+    private flameTimeout?: NodeJS.Timeout;
+    private setIsUpdatingIntelliSense(val: boolean): void {
+
+        const settings: CppSettings = new CppSettings((vscode.workspace.workspaceFolders && vscode.workspace.workspaceFolders.length > 0) ? vscode.workspace.workspaceFolders[0]?.uri : undefined);
+
+        // TODO: Integrate with Tarik's feature to determine if compiler/bare-intellisense is configured
+        if (settings.intelliSenseEngine === "disabled") {
+            this.intelliSenseStatusBarItem.text = this.missingIntelliSenseText;
+            this.intelliSenseStatusBarItem.command = {
+                command: "C_Cpp.CheckForCompilerUI_Telemetry",
+                title: localize("intellisense.select.text", "Select a Compiler")
+            };
+            return;
+        }
+
+        this.intelliSenseStatusBarItem.busy = val;
+
+        if (this.flameTimeout) {
+            clearTimeout(this.flameTimeout);
+        }
+
+        if (val) {
+            this.intelliSenseStatusBarItem.text = "$(flame)";
+            this.intelliSenseStatusBarItem.detail = this.updatingIntelliSenseText;
+        } else {
+            this.flameTimeout = setTimeout(() => {
+                if (this.intelliSenseStatusBarItem) {
+                    this.intelliSenseStatusBarItem.text = this.idleIntelliSenseText;
+                    this.intelliSenseStatusBarItem.detail = "";
+                }
+            }, this.iconDelayTime);
+        }
+        this.intelliSenseStatusBarItem.command = {
+            command: "C_Cpp.RestartIntelliSenseForFileUI_Telemetry",
+            title: localize("rescan.intellisense.text", "Rescan"),
```

Comment:
"Rescan" should be localized.

Reply:
Which line? I don't see it

Reply:
I don't see it anymore either. Maybe I misread it initially.